RESTful Samples/Batch Processing/C#/JsonBatchProcessing/Program.cs

```diff
@@ -0,0 +1,173 @@
+                status = await client.GetDocumentStatus(id);
+                continue;
+            }
+
+            // Document has been generated successfully, time to retrieve it
+            Console.WriteLine("Request for {0} completed...", name);
+            Document doc = await client.GetDocument(id);
+
+            // Save the completed document to disk, in the GeneratedDocs folder of the sample directory
+            string filepath = AppDomain.CurrentDomain.BaseDirectory.Replace("\\bin\\Debug\\netcoreapp3.1\\", "\\GeneratedDocs\\");
+            filepath = Path.GetFullPath(filepath + name + ".pdf");
+            File.WriteAllBytes(filepath, doc.Data);
+
+            // Remove the successfully completed request from the server
+            await client.DeleteDocument(id);
+            return doc;
+        }
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Starting...");
+
+            // Instantiate the client, connecting to the RESTful engine hosted at the given url
+            string url = "http://ec2-34-201-126-96.compute-1.amazonaws.com";
+            WindwardClient client = new WindwardClient(new Uri(url));
+
+            // Set the license key these requests will be ran with
+            // Only required if the server doesn't have it set in the config file
+            // Replace [LICENSE_KEY] with your license key
+            client.LicenseKey = "[LICENSE_KEY]";
+
+            // Get the location of our template, located in a hosted S3 bucket
+            string tempURL = "https://windward-private-bucket.s3.amazonaws.com/LemonadeStandInvoice.docx";
+
+            // Parse our list of records, creating a list of JSON objects corresponding to each order
+            IList<Order> orderList = new List<Order>();
+            try
+            {
+                string dataFile = AppDomain.CurrentDomain.BaseDirectory.Replace("\\bin\\Debug\\netcoreapp3.1\\", "\\Data\\LemonadeStandOrders.json");
+                string records = File.ReadAllText(dataFile);
+                JObject lemonadeOrders = JObject.Parse(records);
+                IList<JToken> orders = lemonadeOrders["Orders"].Children().ToList();
+                foreach (JToken order in orders)
+                {
+                    Order temp = order.ToObject<Order>();
+                    orderList.Add(temp);
+                }
+                Console.WriteLine("Parsed Records...");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+
+            
+            List<Task<string[]>> requests = new List<Task<string[]>>();
+            int InvoiceNum = 1;
+            // Create a task to send a request for each record
+            foreach (Order order in orderList)
+            {
+                // Convert one record into a byte array to be sent to the server
+                byte[] data = System.Text.Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(order));
```

Comment:
Maybe instead of having two forloops have just the initial one. Instead of adding to an order list you can just get the bytes and post it in the first forloop